roles/kube_metrics_server/tasks/main.yml

```diff
@@ -0,0 +1,32 @@
+- name: 'Copy metrics-server deployment files'
+  copy:
+    src: '{{ item }}'
+    dest: '{{ ms_addon_dir }}/'
+    owner: root
+    group: root
+    mode: 0644
+    directory_mode: 0755
+  with_items:
+    - auth-delegator.yaml
+    - auth-reader.yaml
+    - metrics-apiservice.yaml
+    - metrics-server-deployment.yaml
+    - metrics-server-service.yaml
+    - resource-reader.yaml
+  run_once: true
+  delegate_to: "{{ groups['kube-master'][0] }}"
+
+- name: 'Deploy metrics-server'
+  kube:
+    kubectl: '{{ bin_dir }}/kubectl'
+    filename: '{{ ms_addon_dir }}/{{ item }}'
+    state: 'latest'
+  with_items:
```

Comment:
Avoiding repetition of files, could be done via `register: ms_manifests` on template task, 
and `with_items: '{{ ms_manifests.results|default([])|map(attribute='item')|list }}'` on this one